MINIFICPP-365 Adjusting log levels.

apiri opened the pull request:
current status of working to remove extraneous log statements.

for some of the typical flows I use for testing this amounts to almost no information being provided.  going to look through the various components a bit more to see if statements should be added

phrocker commented on libminifi/src/sitetosite/SiteToSiteClient.cpp:
```diff
@@ -467,12 +467,12 @@ int16_t SiteToSiteClient::send(std::string transactionID, DataPacket *packet, co
   }
 
   if (transaction->getState() != TRANSACTION_STARTED && transaction->getState() != DATA_EXCHANGED) {
-    logger_->log_debug("Site2Site transaction %s is not at started or exchanged state", transactionID.c_str());
```
we should also make it a point to change these from transactionID.c_str() to tranactionID

phrocker commented on main/MiNiFiMain.cpp:
```diff
@@ -110,7 +110,7 @@ int main(int argc, char **argv) {
 
   if (signal(SIGINT, sigHandler) == SIG_ERR || signal(SIGTERM, sigHandler) == SIG_ERR || signal(SIGPIPE, SIG_IGN) == SIG_ERR) {
-    logger->log_error("Can not install signal handler");
+    logger->log_error("Cannot install signal handler");
```
Seems unnecessary, both are acceptable spellings.

phrocker commented on libminifi/src/sitetosite/RawSocketProtocol.cpp:
```diff
@@ -519,7 +519,7 @@ bool RawSiteToSiteClient::bootstrap() {
 
   if (establish() && handShake() && negotiateCodec()) {
-    logger_->log_info("Site2Site Ready For data transaction");
+    logger_->log_trace("Site2Site Ready For data transaction");
```
If you were to set the log level to debug one might expect there to be something here, as you would need to select trace and inevitably trace the entire flow of execution vice getting debug information.

phrocker commented on libminifi/src/sitetosite/RawSocketProtocol.cpp:
```diff
@@ -112,11 +112,11 @@ bool RawSiteToSiteClient::initiateResourceNegotiation() {
   int ret = peer_->writeUTF(getResourceName());
 
-  logger_->log_info("result of writing resource name is %i", ret);
+  logger_->log_debug("result of writing resource name is %i", ret);
```
^ this honestly seems like it would be a trace level to help trace the level of execution through the stream writes.

phrocker commented on libminifi/src/core/ProcessSessionReadCallback.cpp:
```diff
@@ -64,16 +64,16 @@ int64_t ProcessSessionReadCallback::process(std::shared_ptr<io::BaseStream> stre
 bool ProcessSessionReadCallback::commit() {
   bool success = false;
 
-  logger_->log_info("committing export operation to %s", _destFile.c_str());
```
ditto on the c_str(). It adds overhead and is vestigial.

phrocker commented on libminifi/src/core/ProcessSession.cpp:
```diff
@@ -568,16 +568,16 @@ void ProcessSession::import(std::string source, std::vector<std::shared_ptr<Flow
 
   if (commit_ok) {
-    logger_->log_info("Commit OK.");
+    logger_->log_debug("Commit OK.");
```
I'm on the fence about this one. I usually use the mantra as INFO provides informational statements for support staff, debug and trace used by developers for varying levels. In this case I could see a support staff asking, "Commit OK."  Is there an analog to this statement to provide this information? A totally quiet client at info makes it unusable and more importantly unsupportable by multiple levels of support.

phrocker commented on libminifi/src/controllers/SSLContextService.cpp:
```diff
@@ -134,7 +134,7 @@ void SSLContextService::onEnable() {
         logger_->log_debug("%s now good", certificate);
       } else {
-        logger_->log_debug("%s still not good", test_cert);
+        logger_->log_warn("%s still not good", test_cert);
```
good catch!

phrocker commented on libminifi/src/capi/Plan.cpp:
```diff
@@ -115,7 +115,7 @@ bool linkToPrevious) {
 bool ExecutionPlan::setProperty(const std::shared_ptr<core::Processor> proc, const std::string &prop, const std::string &value) {
   uint32_t i = 0;
-  logger_->log_info("Attempting to set property %s %s for %s", prop, value, proc->getName());
+  logger_->log_debug("Attempting to set property %s %s for %s", prop, value, proc->getName());
```
oh man this is definitely me! I copied and pasted a lot of these. Thanks for catching this!

phrocker commented on libminifi/src/ThreadedSchedulingAgent.cpp:
```diff
@@ -104,10 +104,10 @@ void ThreadedSchedulingAgent::stop() {
+  logger_->log_debug("Shutting down threads for processor %s/%s", processor->getName().c_str(), processor->getUUIDStr().c_str());
 
   if (processor->getScheduledState() != core::RUNNING) {
-    logger_->log_info("Cannot unschedule threads for processor %s because it is not running", processor->getName().c_str());
```
Much prefer this!

phrocker commented on libminifi/src/SchedulingAgent.cpp:
```diff
@@ -40,7 +40,7 @@ bool SchedulingAgent::hasWorkToDo(std::shared_ptr<core::Processor> processor) {
 }
 
 std::future<bool> SchedulingAgent::enableControllerService(std::shared_ptr<core::controller::ControllerServiceNode> &serviceNode) {
-  logger_->log_info("Enabling CSN in SchedulingAgent %s", serviceNode->getName());
```
As per my comment below I think these are candidates for INFO based on the belief that support any type of teir 1 support might need this information.

phrocker commented on libminifi/src/RemoteProcessorGroupPort.cpp:
```diff
@@ -193,17 +193,17 @@ void RemoteProcessorGroupPort::onTrigger(const std::shared_ptr<core::ProcessCont
     protocol_ = getNextProtocol();
 
     if (!protocol_) {
-      logger_->log_info("no protocol, yielding");
```
This one should likely be debug otherwise we'd need to get to trace to support this. The one above is definitely a trace.

phrocker commented on libminifi/src/FlowControlProtocol.cpp:
```diff
@@ -216,7 +216,7 @@ void FlowControlProtocol::stop() {
     return;
   running_ = false;
-  logger_->log_info("FlowControl Protocol Stop");
+  logger_->log_trace("FlowControl Protocol Stop");
```
I believe these messages provide critical information to tier1 support.

phrocker commented on libminifi/include/core/controller/StandardControllerServiceProvider.h:
```diff
@@ -111,10 +111,10 @@ class StandardControllerServiceProvider : public ControllerServiceProvider, publ
     for (auto service : controller_map_->getAllControllerServices()) {
 
       if (service->canEnable()) {
-        logger_->log_info("Enabling %s", service->getName());
```
Ditto on the comments about enabling/starting something be critical for tier 1 to support.

phrocker commented on extensions/http-curl/sitetosite/HTTPProtocol.cpp:
```diff
@@ -256,7 +256,7 @@ void HttpSiteToSiteClient::closeTransaction(const std::string &transactionID) {
   }
 
   std::string append_str;
-  logger_->log_info("Site2Site close transaction %s", transaction->getUUIDStr().c_str());
```
I think this is truly informative for teir 1 support.

apiri commented:
@phrocker I agree.  I'm going to take a second pass over this and with your feedback try to get a bit more info ready.  will push out another commit

apiri commented:
@phrocker I incorporated your feedback, added logs to session transfers/penalization and adjusted the logs for RPG send/receive.  I think it strikes a decent balance of showing that work is going on with pertinent details.  Let me know your thoughts.
